Library/Homebrew/formula.rb

```diff
@@ -3201,6 +3201,9 @@ def skip_clean(*paths)
     # <pre>keg_only :versioned_formulae</pre>
     # <pre>keg_only "because I want it so"</pre>
     def keg_only(reason, explanation = "")
+      if !Homebrew::EnvConfig.no_install_from_api? && reason.is_a?(String)
+        reason = reason.gsub(HOMEBREW_PREFIX, HOMEBREW_PREFIX_PLACEHOLDER)
```

Comment:
Oh, I misread this part. This should go in `Formula#to_hash`. What's happening is you're writing the correctly-substituted string to `keg_only` when loading from the API, but then you're immediately replacing it back with the placeholder before actually creating the `keg_only_reason`.

Replacing with the placeholder only needs to happen when we're actually creating the JSON.

Reply:
Oh! 😓 Thanks!

Reply:
🤔 I did see that the caveats in `to_hash` already had the substitution there, but caveats is a string so we can do that directly. But we can't with `keg_only_reason` since it has `reason` and `explanation`, right?

Reply:
Wait, no, ignore me, I misunderstood.

Reply:
Done.